pkg/rtc/rtc_linux_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rtc
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+func TestSet(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+	}{
+		{
+			name: "success",
+		},
+		{
+			name: "error",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := ioutil.TempFile("", "rtc-*")
+			if err != nil {
+				t.Error(err)
+			}
+			devs = []string{f.Name()}
+			f.Close()
+			rtc, err := OpenRTC()
+			if err != nil {
+				t.Error(err)
+			}
+			if tt.name == "success" {
+				unixIoctlSetRTCTime = func(fd int, value *unix.RTCTime) error {
+					return nil
+				}
+				err = rtc.Set(time.Now())
+				if err != nil {
+					t.Error(err)
+				}
+			}
+			if tt.name == "error" {
+				unixIoctlSetRTCTime = func(fd int, value *unix.RTCTime) error {
+					return errors.New("error")
+				}
+				err = rtc.Set(time.Now())
+				if err == nil {
+					t.Error(err)
+				}
+			}
+			rtc.Close()
+			err = os.Remove(devs[0])
+			if err != nil {
+				t.Error(err)
+			}
+			unixIoctlSetRTCTime = unix.IoctlSetRTCTime
+		})
+	}
+}
+
+func TestRead(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+	}{
+		{
+			name: "success",
+		},
+		{
+			name: "error",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := ioutil.TempFile("", "rtc-*")
+			if err != nil {
+				t.Error(err)
+			}
+			devs = []string{f.Name()}
+			f.Close()
```

Comment:
it's common to have a 
defer os.Remove(f.Name())
so we don't litter /tmp with these files.
This is a failing of Unix and its share /tmp, but it's all we have ...

Reply:
Same here, instead of running defer I manually called os.Remove on it further down in the form of os.Remove(devs[0]) as the devs array only holds a single string that is f.Name()

Reply:
For sure this works here. Using defer is just good practice. For this code it does not matter.